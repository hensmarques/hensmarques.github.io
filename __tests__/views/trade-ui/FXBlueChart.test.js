import React from 'react';
import { FXBlueChart } from '../../../src/views/trade-ui/FXBlueChart';

describe('<FXBlueChart/>', () => {
  // mocks for finding and manipulating chart on document
  const postMessageMock = jest.fn();
  document.getElementById = jest.fn().mockImplementation(id => {
    if (id === 'fxblue-chart') {
      return {
        contentWindow: {
          postMessage: postMessageMock
        }
      };
    }
  });
  const selectedInstrument = {
    id: 'BTCGBP',
    base: 'BTC',
    quote: 'GBP',
    quoteDecimals: 3
  };
  const setFxBlueChartTimeFrame = jest.fn();
  const loadSelectedFxBlueChartTimeFrame = jest.fn();
  const selectedTimeframe = '1Y';
  const propsMock = {
    config: {
      siteName: 'test',
      chartConfig: {
        name: 'oms-d-demo'
      }
    },
    user: 'test',
  }
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FXBlueChart 
      selectedInstrument={selectedInstrument}
      setFxBlueChartTimeFrame={setFxBlueChartTimeFrame}
      loadSelectedFxBlueChartTimeFrame={loadSelectedFxBlueChartTimeFrame}
      {...propsMock}
      />);
  });

  it('renders and matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(loadSelectedFxBlueChartTimeFrame).toHaveBeenCalled();
  });

  it('posts the appropriate message to the chart', () => {
    expect(document.getElementById).toHaveBeenCalledWith('fxblue-chart');
    expect(postMessageMock).toHaveBeenCalledWith(
      { command: 'setChartSymbol', symbol: 'bitfinex@BTCGBP' },
      '*'
    );
  });
});
