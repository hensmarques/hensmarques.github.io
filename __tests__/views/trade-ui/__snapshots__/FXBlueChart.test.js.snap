// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FXBlueChart/> renders and matches the snapshot 1`] = `
<div
  className="fxblue-chart"
>
  <div
    className="chart-timeframe-selector"
  >
    <span
      className="chart-timeframe-option "
      key="0"
      onClick={[Function]}
    >
      1H
    </span>
    <span
      className="chart-timeframe-option "
      key="1"
      onClick={[Function]}
    >
      1D
    </span>
    <span
      className="chart-timeframe-option active"
      key="2"
      onClick={[Function]}
    >
      1W
    </span>
    <span
      className="chart-timeframe-option "
      key="3"
      onClick={[Function]}
    >
      1M
    </span>
    <span
      className="chart-timeframe-option "
      key="4"
      onClick={[Function]}
    >
      3M
    </span>
    <span
      className="chart-timeframe-option "
      key="5"
      onClick={[Function]}
    >
      1Y
    </span>
  </div>
  <ShiftIframe
    allowFullScreen={false}
    height="400px"
    id="fxblue-chart"
    name=""
    src="https://shiftforex.fxbluelabs.com/?css=//localhost/fx-blue-charts.css&bitfinex@BTCGBP&exchangeID=undefined&userID="
    width="100%"
  >
    <p>
      This feature requires iframe-support to display.
    </p>
  </ShiftIframe>
</div>
`;
